sys/event/timeout_ztimer.c

```diff
@@ -29,6 +29,9 @@ void event_timeout_init(event_timeout_t *event_timeout, event_queue_t *queue, ev
 void event_timeout_ztimer_init(event_timeout_t *event_timeout, ztimer_clock_t *clock,
                                event_queue_t *queue, event_t *event)
 {
+    if (event_timeout->clock) {
```

Comment:
hmm. on a random (stack allocated) event_timeout, this will break badly, I think.